include/mitsuba/render/sunsky.h

```diff
@@ -0,0 +1,610 @@
+/*
+ * This file contains various helpers for the `sunsky` plugin.
+ */
+
+#pragma once
+
+#include <mitsuba/core/fresolver.h>
+#include <mitsuba/core/fstream.h>
+#include <mitsuba/core/spectrum.h>
+#include <drjit/sphere.h>
+
+
+NAMESPACE_BEGIN(mitsuba)
+
+/// Number of spectral channels in the skylight model
+static const constexpr size_t WAVELENGTH_COUNT = 11;
+/// Number of turbidity levels in the skylight model
+static const constexpr size_t TURBITDITY_LVLS = 10;
+/// Number of albedo levels in the skylight model
+static const constexpr size_t ALBEDO_LVLS = 2;
+
+/// Distance between wavelengths in the skylight model
+static constexpr size_t WAVELENGTH_STEP = 40;
+/// Wavelengths used in the skylight model
+template <typename Float>
+static constexpr Float WAVELENGTHS[WAVELENGTH_COUNT] = {
+    320, 360, 400, 440, 480, 520, 560, 600, 640, 680, 720
+};
+
+/// Number of control points for interpolation in the skylight model
+static const constexpr size_t SKY_CTRL_PTS = 6;
+/// Number of parameters for the skylight model equation
+static const constexpr size_t SKY_PARAMS = 9;
+
+/// Number of control points for interpolation in the sun model
+static const constexpr size_t SUN_CTRL_PTS = 4;
+/// Number of segments for the piecewise polynomial in the sun model
+static const constexpr size_t SUN_SEGMENTS = 45;
+/// Number of coefficients for the sun's limb darkening
+static const constexpr size_t SUN_LD_PARAMS = 6;
+
+/// Number of elevation control points for the tgmm sampling tables
+static const constexpr size_t ELEVATION_CTRL_PTS = 30;
+/// Number of gaussian components in the tgmm
+static const constexpr size_t TGMM_COMPONENTS = 5;
+/// Number of parameters for each gaussian component
+static const constexpr size_t TGMM_GAUSSIAN_PARAMS = 5;
+
+/// Sun half aperture angle in radians
+#define SUN_HALF_APERTURE (dr::deg_to_rad(0.5358/2.0))
+/// Mean radius of the Earth
+static const constexpr double EARTH_MEAN_RADIUS = 6371.01;   // In km
+/// Astronomical unit
+static const constexpr size_t ASTRONOMICAL_UNIT = 149597890; // In km
+
+/// Conversion constant to convert spectral solar luminosity to RGB
+static const constexpr float SPEC_TO_RGB_SUN_CONV = 467.069280386;
+
+#define DATABASE_PATH "data/sunsky/"
+
+enum class Dataset {
+    SkyParams, SkyRadiance,
+    SunRadiance, SunLimbDarkening,
+    TGMMTables
+};
+
+// ================================================================================================
+// ====================================== HELPER FUNCTIONS ========================================
+// ================================================================================================
+
+/**
+ * \brief Converts a spherical unit vector to it's angles
+ *
+ * \param v
+ *      Vector to convert
+ * \return
+ *      The {phi, theta} angles
+ */
+template <typename Value>
+MI_INLINE Point<Value, 2> from_spherical(const Vector<Value, 3> &v) {
+    return { dr::atan2(v.y(), v.x()), dr::unit_angle_z(v) };
+}
```

Comment:
I keep re-implementing this function. Would it make sense to move it to `math.h` instead and expose it with Python bindings?

Reply:
I was convinced we had these in the new Dr.Jit, but I can't find them. I'll make bindings for them in Mitsuba :+1: